src/account/UpgradeableModularAccount.sol

```diff
@@ -262,6 +262,41 @@ contract UpgradeableModularAccount is
         return returnData;
     }
 
+    /// @inheritdoc IPluginExecutor
+    function executeWithAuthorization(bytes calldata data, bytes calldata authorization)
+        external
+        payable
+        returns (bytes memory)
+    {
+        bytes4 execSelector = bytes4(data[0:4]);
+
+        FunctionReference runtimeValidationFunction = FunctionReference.wrap(bytes21(authorization[0:21]));
```

Comment:
Nit: do we allow invalid `runtimeValidationFunction`? If not, should we consider validating here before reading into storage?

Reply:
Could you please explain your question a bit more? The workflow here is currently:
- The requested validation function is read from the data in `authorization`.
-  The account checks to make sure that this validation function is actually installed.
- The account checks that the validation function applies to the selector this call is trying to run.

Reply:
ah I just meant function references like `bytes21(0)` might not be valid, so we can probably fail fast here

Reply:
Ah I see, since that step will fail anyways on the storage check later I don't know if it's worth failing early - it reduces gas in the failure case but increases gas in the success case, and generally I think optimizing for the success case is better.